### PR TITLE
Upgrade step for dynamic address configs

### PR DESCRIPTION
**Pull request by manadart**
In a prior patch we unified the type for address configuration method. This meant that prior usage of "dynamic" from the the removed type values was replaced with "dhcp" under the unified values.

This upgrade step replaces _config-method_ "dynamic" with "dhcp" in the _ip.addresses_ collection.

Beyond integrity of data, no behaviour is changed - _config-method_ is currently only ever compared with "loopback".

## QA steps

- Bootstrap < 2.9.5 to LXD.
- `juju add-machine`.
- When the machine comes up, `juju ssh 0` and stop it's agent so we don't get updated link-layer data: `systemctl stop jujud-machine-0`.
- Connect to Mongo. There should only be one address in the workload model that is not for a loopback device. Update it's config method to be "dynamic" for example:
```
db.ip.addresses.updateOne({"_id" : "cfd25afa-af5c-4de2-8a7e-ba6ca5c0a397:m#0#d#eth0#ip#10.161.87.115"}, {$set: {"config-method": "dynamic"}})
```
- Upgrade to this patch and observe that the config method is changed to "dhcp".

## Documentation changes

None.

## Bug reference

N/A

### CODE REVIEW

**Comment by manadart** [comment]
[Upgrade step for DHCP address config method](https://trello.com/c/qkshALvq/3699-upgrade-step-for-dhcp-address-config-method)

File: state/upgrades.go [hunk 1]
```diff
@@ -3482,3 +3482,31 @@ func TranslateK8sServiceTypes(pool *StatePool) error {
...
+	for iter.Next(&doc) {
+		ops = append(ops, txn.Op{
+			C:      ipAddressesC,
+			Id:     doc["_id"],
+			Assert: txn.DocExists,
+			Update: bson.M{"$set": bson.M{"config-method": network.ConfigDHCP}},
+		})
+	}
+
+	if len(ops) > 0 {
```
Comment (achilleasa) [inline state/upgrades.go:3508]: You probably need to add a:

```
if err := iter.Close(); err != nil {
  return errors.Trace(err)
}
```

before checking `len(ops)`.

**Review by achilleasa — APPROVED** [review]
QA works but please make sure to close the iterator before you land the PR!

File: state/upgrades.go (hunk 1 above)
Comment (manadart) [reply state/upgrades.go:3508]: Good catch; ta.

**Comment by manadart** [comment]
$$merge$$

**Pull request merged**